lib/cli-manager.js

```diff
@@ -0,0 +1,352 @@
+        debugger;
+        this._processCommonOptions(argv);
+        this._processAuthOptions(argv);
+
+        this._loadGlobalSetup();
+
+        if (!this._isOneTimeSession && this._requiresAuth) {
+          this._setCurrentUserFromProfile(argv[AuthOptionsNames.PROFILE]);
+        }
+
+        if (this._requiresAuth && !this._isOneTimeSession && !this._authentication.hasCurrentUser()) {
+          // TODO: change err msg
+          throw new Error('You must be authenticated. (CHANGE ERR MSG)');
+        }
+
+        return true;
+      })
+      .command(cmdInit(this._authController))
+      .command('flex', 'Deploy and manage flex services. Run \'kinvey flex -h\' for details.', cmdFlex(this._flexController))
+      .demand(1, '')
+      .strict(true)
+      .help('h')
+      .alias('h', 'help')
+      .showHelpOnFail(true);
+
+    yargs.argv;
+  }
+
+  init(args) {
+    this._initCommandsManager(args);
+  }
+
+  sendRequest(options, done) {
+    if (!options.host) {
+      options.host = this.config.host;
+    }
+
+    const reqObj = new Request(this._authentication.getCurrentUser(), options);
+
+    this.log(LogLevel.DEBUG, 'Request:  %s %s', reqObj.options.method, reqObj.options.url);
+
+    return reqObj.send((err, res) => {
+      this.log(LogLevel.DEBUG, 'Response: %s %s %s', reqObj.options.method, reqObj.options.url, chalk.green(res.statusCode));
+      done(err, res.body);
+    });
+  }
+
+  log(logLevel, ...args) {
+    this._logger[logLevel](...args);
+  }
+
+  _clearSession(done) {
+    if(!this._isOneTimeSession) {
+      return done(null);
+    }
+
+
+    this._authentication.logout(done);
+  }
+
+  processCommandResult(err, result, done) {
+    this._clearSession(() => {
+      const isCallbackProvided = typeof done === 'function';
+      if (err) {
+        this.log(LogLevel.ERROR, '%s', err);
+
+        if (isCallbackProvided) {
+          return done(err);
+        }
+
+        process.exit(1);
+      } else {
+        if (isCallbackProvided) {
+          return done(null, result);
+        }
+      }
+    });
+  }
+
+  /**
+   * Sets the current user if not already set using info from options.
+   * @param {Object} options
+   * @param done
+   * @returns {*}
+   */
+  setCurrentUserFromOptions(options, done) {
+    if (this._authentication.hasCurrentUser()) {
+      return done(null);
+    }
+
+    this.login(options[AuthOptionsNames.EMAIL], options[AuthOptionsNames.PASSWORD], null, options[AuthOptionsNames.HOST], done);
+  }
+
+  login(email, password, MFAToken, host = this.config.host, done) {
+    this._authentication.login(email, password, MFAToken, host, done);
+  }
+
+  createProfile(profileName, email, password, MFAToken, host, done) {
+    // TODO: decide what to do if profile already exists
+    if (utils.isNullOrUndefined(profileName)) {
+      return done(new Error('Profile name is not set.'));
+    }
+
+    const self = this;
```

Comment:
Is there a specific reason for using `const self = this` instead of using arrow functions in the `async.series` array?

Reply:
I just wanted to give names to the functions executed by async, so that it's clear what's going on.

Reply:
I'd use comments for that purpose then, to avoid saving references to `this` unless it's absolutely needed (for example, in nock): https://github.com/Kinvey/style-guides/blob/master/backend/es6.md#naming--self-this